### PR TITLE
use pypi upstream, run tests, pip check, update metadata

### LINKED ISSUE #45: dlib doesn't pass pip check
Thanks for maintaining this package!

I've run into an [issue](https://github.com/conda-forge/lipo-feedstock/pull/1#issuecomment-667608963) when trying to validate an environment containing a package that depends on `dlib` with `pip check`.

Unfortunately, I don't know much about this package. This appears to be a rather involved build, to be sure, but the contents of the distribution look substantially different than, say, the pypi source distribution.

I see there is also `dlib-cpp`. 

Any thoughts on what might be causing this? Are there additional steps that could be taken to make this appear more "normal" to python tools?

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Notes:
- fixes #45
- unblocks https://github.com/conda-forge/lipo-feedstock/pull/1
- uses the upstream source distribution from pypi
- adds a zero to the version number, but bumps the build number anyway
- runs the tests included in the distribution with `pytest(-azurepipelines)`
  - not all of the test fixtures are included
  - couldn't get coverage to report anything
- adds a patch to push the custom args to `cmake` via environment variable
  - i confess, i don't know what's all going on, just reading the build tea leaves
- uses pip check (not that dlib declares any of its dependencies, sigh)
- updates some metadata
- hasn't tested this on windows yet, so we'll see what CI thinks of my hacks

I'm not _super_ interested in another feedstock to maintain, as I'm not a direct user, but can add myself as a maintainer if that's the price to get it packaging a little better for downstreams!

### CODE REVIEW

**Comment by bollwyvl** [comment]
Closing in on it... needs something like:
https://github.com/conda-forge/pyscal-feedstock/pull/19/files#diff-871abb66a0fee5e1303b147625701f13

**Comment by bollwyvl** [comment]
@conda-forge/dlib ready for review!

**Comment by bollwyvl** [comment]
The tests are throwing a lot of spurious warnings: working on a fix...

**Comment by bollwyvl** [comment]
Well, I don't think I can fix the remaining warnings about the paths, as the build so graciously rewrites our paths for us, but it is a little cleaner (and encapsulates the test setup a little better). The test reports _do_ actually get uploaded, it seems, as the _Tests_ tab is populated.

Aside from maybe debugging (read: rekicking) that dangling OSX build, that's about all I'm planning to do, thanks again!

**Comment by patricksnape** [comment]
Yeah so this awesome. For a long time the dlib builds didn't have any proper Python packaging and then there was a point where the Python packaging was just fighting against conda everywhere and constantly not correctly linking packages. Since all the tests pass and I can't see any real complaints about incorrect linking I think this is OK.

Can we maybe just remove the commented out text from the shell scripts and then we can merge?

Thanks again for this!

File: recipe/build.sh [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
-  -DCMAKE_BUILD_TYPE=Release  \
-  -DPYTHON_EXECUTABLE=${PREFIX}/bin/python \
-  -DJPEG_INCLUDE_DIR=${PREFIX}/include \
-  -DPNG_PNG_INCLUDE_DIR=${PREFIX}/include \
-  -DPNG_INCLUDE_DIR=${PREFIX}/include \
-  -Dsqlite_path=${PREFIX}/include \
-  ../tools/python
-make -j$CPU_COUNT
-cp dlib*.so $SP_DIR
+# previous script for posterity
```
Comment (patricksnape) [inline recipe/build.sh:6]: This is all in the git history so let's just remove it

File: recipe/bld.bat [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
-  -DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%"   ^
-  -DDLIB_LINK_WITH_SQLITE3=OFF                ^
-  -DDLIB_FORCE_MSVC_STATIC_RUNTIME=OFF        ^
-  ../tools/python
-if errorlevel 1 exit 1
+REM mkdir build && cd build
 
-cmake --build .
-if errorlevel 1 exit 1
+REM cmake -LAH -G"NMake Makefiles"                ^
```
Comment (patricksnape) [inline recipe/bld.bat:9]: This is all in the git history so let's just remove it

Comment (bollwyvl) [reply recipe/bld.bat]: 42d9523

File: recipe/build.sh (hunk 1 above)
Comment (bollwyvl) [reply recipe/build.sh]: 42d9523

**Comment by bollwyvl** [comment]
Thanks for the speedy review! 

I did another pass over the scripts for readability, and added a few comments and log output.

Regarding the historical build woes: yeah, as a non-user of this, I'm just reacting to how it builds, actually seems pretty smooth. And my goodness, hooray for first-party tests. On that note: I'll find out how this plays for sure when I fire `dlib` (by way of `lipo`) up along the with the other dozens of optional dependencies in the test suite of the frequently-released `diffoscope`, as they're definitely expecting things to work The Debian Way, which can often uncover issues we've made for ourselves on conda-forge.

**Comment by bollwyvl** [comment]
I have to head away from the keyboard for a while but I'm seeing some green in the CI roster: feel free to tweak/fork/squash whatever you find here!

**Comment by patricksnape** [comment]
No I have confidence in the current tests - so I am going to merge this.

Thanks again for sorting this out!

**Comment by bollwyvl** [comment]
Great, thanks again!

**Pull request merged**